beta/nncf/tensorflow/pruning/base_algorithm.py

```diff
@@ -0,0 +1,271 @@
+from nncf.common.utils.logger import logger as nncf_logger
+
+
+class PrunedLayerInfo:
+    BN_LAYER_NAME = 'bn_layer'
+
+    def __init__(self, layer_name: str, related_layers: Dict[str, List[str]]):
+        self.layer_name = layer_name
+        self.related_layers = related_layers
+
+
+class BasePruningAlgoBuilder(TFCompressionAlgorithmBuilder):
+
+    def __init__(self, config):
+        super().__init__(config)
+        params = config.get('params', {})
+        self._params = params
+
+        self._ignore_frozen_layers = True
+        self._prune_first = params.get('prune_first_conv', False)
+        self._prune_last = params.get('prune_last_conv', False)
+        self._prune_batch_norms = params.get('prune_batch_norms', True)
+        self._prune_downsample_convs = params.get('prune_downsample_convs', False)
+
+        self._prunable_types = self._get_op_types_of_pruned_layers()
+        types_of_grouping_ops = self._get_types_of_grouping_ops()
+        self._pruning_node_selector = PruningNodeSelector(TF_PRUNING_OPERATOR_METATYPES,
+                                                          self._prunable_types,
+                                                          types_of_grouping_ops,
+                                                          None, # self.ignored_scopes,
+                                                          None, # self.target_scopes,
+                                                          self._prune_first,
+                                                          self._prune_last,
+                                                          self._prune_downsample_convs)
+
+        self._pruned_layer_groups_info = None
+
+    def apply_to(self, model: tf.keras.Model) -> tf.keras.Model:
+        transformation_layout = self.get_transformation_layout(model)
+        return TFModelTransformer(model, transformation_layout).transform()
+
+    def get_transformation_layout(self, model: tf.keras.Model) -> TFTransformationLayout:
+        graph = convert_keras_model_to_nncf_graph(model)
+        groups_of_nodes_to_prune = self._pruning_node_selector.create_pruning_groups(graph)
+
+        transformations = TFTransformationLayout()
+        shared_layers = set()
+
+        self._pruned_layer_groups_info = Clusterization('layer_name')
+
+        for i, group in enumerate(groups_of_nodes_to_prune.get_all_clusters()):
+            group_minfos = []
+            for node in group.nodes:
+                layer_name = tf_get_layer_identifier(node)
+                if layer_name in shared_layers:
+                    continue
+                if is_shared(node):
+                    shared_layers.add(layer_name)
+                layer = model.get_layer(layer_name)
+                # Check that we need to prune weights in this op
+                assert self._is_pruned_layer(layer)
+
+                nncf_logger.info("Adding Weight Pruner in: {}".format(layer_name))
```

Comment:
[layer_name should be a logger argument](https://github.com/openvinotoolkit/nncf/blob/develop/docs/styleguide/PyGuide.md#37-logging)

Reply:
also "" -> ''

Reply:
Fixed